Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/StitchByBackground.py

```diff
@@ -0,0 +1,150 @@
+            errors["InputWorkspaces"] = err_msg
+            errors["StitchPoints"] = err_msg
+
+        upper_bound = self.getProperty("CropUpperBound").value
+        lower_bound = self.getProperty("CropLowerBound").value
+        if not upper_bound > lower_bound:
+            err_msg = f"Upper bound ({upper_bound}) must be greater than lower bound ({lower_bound})."
+            errors["CropUpperBound"] = err_msg
+            errors["CropLowerBound"] = err_msg
+        return errors
+
+    def PyInit(self):
+        self.declareProperty(
+            StringArrayProperty("InputWorkspaces", "", direction=Direction.Input, validator=ADSValidator()),
+            doc="List of workspaces to be stitched together.",
+        )
+        self.declareProperty(WorkspaceProperty("OutputWorkspace", "", direction=Direction.Output), doc="The stitched workspace.")
+        self.declareProperty(
+            FloatArrayProperty("StitchPoints", [], direction=Direction.Input),
+            doc="Values in Q space where the stitch should take place. i.e. Overlaps in the data.",
+        )
+        self.declareProperty(
+            "OverlapWidth",
+            0.05,
+            validator=FloatBoundedValidator(lower=0),
+            doc="The extent to which the fit limits extend from the points given in StitchPoints.",
+        )
+        self.declareProperty("CropLowerBound", 0.0, doc="The XMin in Q-space to use when cropping the output workspace.")
+        self.declareProperty("CropUpperBound", 0.0, doc="The XMax in Q-space to use when cropping the output workspace.")
+
+    def PyExec(self):
+        ws_name_list = self.getProperty("InputWorkspaces").value
+        ws_list = [AnalysisDataService.retrieve(ws_name) for ws_name in ws_name_list]
+
+        stitch_locations = self.getProperty("StitchPoints").value
+
+        offsets = self.create_offsets(ws_list, stitch_locations)
+
+        subtracted_ws, x, y, e = self.subtract_background(ws_list, offsets, stitch_locations)
+
+        out_ws_name = self.getProperty("OutputWorkspace").value
+        stitched_ws = CreateWorkspace(
+            x,
+            y,
+            e,
+            OutputWorkspace=out_ws_name,
+            UnitX=subtracted_ws.getAxis(0).getUnit().unitID(),
+            Distribution=subtracted_ws.isDistribution(),
+            ParentWorkspace=subtracted_ws,
+            EnableLogging=False,
+        )
+
+        stitched_ws = self.crop_output(stitched_ws)
+
+        self.setProperty("OutputWorkspace", stitched_ws)
+
+    def create_offsets(self, workspaces, stitch_points):
+        overlap_width = self.getProperty("OverlapWidth").value
+        offsets = np.zeros(len(workspaces))
+        for stitch_index, stitch_point in enumerate(stitch_points):
+            backgrounds = []
+            # Get backgrounds for each side of the stitch.
+            for side in range(2):
+                ws = workspaces[stitch_index + side]
+                i_bin_of_stitch = ws.yIndexOfX(stitch_point)
```

Comment:
`yIndexOfX` returns an `IndexError` if the x value is outside the range of the data - technically this could be checked ahead of time in `validateInputs`?